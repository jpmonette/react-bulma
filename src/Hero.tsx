import classNames from "classnames";
import * as React from "react";

interface HeroBodyProps {
  className?: string;
  children?: Element | JSX.Element | string;
}

const HeroBody = ({ children, className, ...props }: HeroBodyProps) => (
  <div {...props} className={classNames(className, "hero-body")}>
    {children}
  </div>
);

interface HeroProps {
  className?: string;
  children?: React.ReactNode;
  isPrimary?: boolean;
  isInfo?: boolean;
  isSuccess?: boolean;
  isWarning?: boolean;
  isDanger?: boolean;
  isLight?: boolean;
  isDark?: boolean;
  isBold?: boolean;
  isMedium?: boolean;
  isLarge?: boolean;
  isFullHeight?: boolean;
}

class Hero extends React.Component<HeroProps> {
  static Body = HeroBody;

  render() {
    const {
      children,
      className,
      isPrimary,
      isInfo,
      isSuccess,
      isWarning,
      isDanger,
      isLight,
      isDark,
      isBold,
      isMedium,
      isLarge,
      isFullHeight,
      ...props
    } = this.props;

    return (
      <section
        {...props}
        className={classNames(
          className,
          "hero",
          isPrimary && "is-primary",
          isInfo && "is-info",
          isSuccess && "is-success",
          isWarning && "is-warning",
          isDanger && "is-danger",
          isLight && "is-light",
          isDark && "is-dark",
          isBold && "is-bold",
          isMedium && "is-medium",
          isLarge && "is-large",
          isFullHeight && "is-fullheight",
        )}
      >
        {children}
      </section>
    );
  }
}
export { Hero };
