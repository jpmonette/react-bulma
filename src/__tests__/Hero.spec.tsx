import * as React from "react";
import { Hero, HeroBody } from "../Hero";
import { create } from "react-test-renderer";

describe("Hero", () => {
  it("should generate a hero", () => {
    const tree = create(
      <Hero
        isPrimary
        isInfo
        isSuccess
        isWarning
        isDanger
        isLight
        isDark
        isBold
        isMedium
        isLarge
        isFullHeight
        className={"test-class"}
      >
        <HeroBody>Sample Website Title</HeroBody>
      </Hero>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
