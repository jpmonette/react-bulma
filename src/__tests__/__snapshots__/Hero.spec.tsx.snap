// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should generate a hero 1`] = `
<section
  className="test-class hero is-primary is-info is-success is-warning is-danger is-light is-dark is-bold is-medium is-large is-fullheight"
>
  <div
    className="hero-body"
  >
    Sample Website Title
  </div>
</section>
`;
